src/Time/Slab.hpp

```diff
@@ -0,0 +1,137 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class Slab
+
+#pragma once
+
+#include <functional>
+#include <limits>
+
+#include "ErrorHandling/Assert.hpp"
+
+class Time;
+class TimeDelta;
+
+namespace PUP {
+class er;
+}
+
+/// A chunk of time.  Every element must reach slab boundaries
+/// exactly, no matter how it actually takes time steps to get there.
+/// The simulation can only be assumed to have global data available
+/// at slab boundaries.
+class Slab {
+ public:
+  /// Default constructor gives an invalid Slab.
+  Slab() noexcept
+      : start_(std::numeric_limits<double>::signaling_NaN()),
+        end_(std::numeric_limits<double>::signaling_NaN()) {}
+
+  /// Construct a slab running between two times (exactly).
+  Slab(double start, double end) noexcept : start_(start), end_(end) {
+    ASSERT(start_ < end_, "Backwards Slab");
+  }
+
+  /// Construct a slab with a given start time and duration.  The
+  /// actual duration may differ by roundoff from the supplied value.
+  static Slab new_with_duration(double start, double duration) noexcept {
```

Comment:
This replies to a few comments:

The overall suggestions are `new_with_*` -> `with_*` and `with_*` -> `create_with_*`.  (The other portions of the suggestions I like and will change.)  This is reversing my original naming scheme and seems very confusing to me (new and create are approximately synonyms).  In particular, I expect methods starting with "create" to be factory-like, and either be static methods or methods on some sort of factory object.

`new_with_` -> `with_` seems okay, except that it depends on the change I don't like.